src/reader.rs

```diff
@@ -6511,10 +6511,22 @@ impl<'a> Reader<'a> {
             prefix.push_utfstr(&tok);
             prefix.push_utfstr(&common_prefix);
         } else {
-            // Append just the end of the string.
+            // Collapse parent directories and append end of string
             prefix.push(get_ellipsis_char());
+
             let full = tok + &common_prefix[..];
-            prefix.push_utfstr(&full[full.len() - PREFIX_MAX_LEN..]);
+            let truncated = &full[full.len() - PREFIX_MAX_LEN..];
+            let parts: Vec<&wstr> = truncated.split('/').collect();
+            if parts.len() < 2 {
```

Comment:
Seems like a good change. I'll try it for a while.
I haven't re-read the issue yet - maybe we can briefly summarize the reason for this change.

I gues zsh will only show the basename even if there is no truncation going on,
but we don't need to go there - your approach is safer.

Reply:
after re-reading the issue, I think we should eventually have a generic way
of making sure that the common prefix is not way larger than any of the suffixes.

But that can definitely be done later, and I expect that we'll still want to keep the special handling of "/".